docs/javascript/v2/how--to-guides/record-and-live-stream/hls/hls-timed-metadata.mdx

```diff
@@ -0,0 +1,116 @@
+	payload: string; // limited to 100 chars, no " (double quotes) allowed
+	duration: number;
+}
+
+sendHLSTimedMetadata(metadatalist: HLSTimedMetadata[]): Promise<void>;
+```
+
+## Receiving HLS Timed Metadata
+### Receiving Payload structure
+
+The payload is sent in the EXT-X-DATERANGE tag as value for the custom attribure X-100MSLIVE-PAYLOAD. It is base64 encoded string of
+
+```json
+{
+    "start_date": "UTC date",
+    "payload": "payload sent by the user",
+    "end_date": "<start_date + duration>"
+    "hms_version": "v1.1"
+}
+```
+
+### APP side implementation
+
+```javascript
+const handleTimeUpdateListener = _ => {
+  const textTrackListCount = videoEl.textTracks.length;
+  let metaTextTrack;
+  for (let trackIndex = 0; trackIndex < textTrackListCount; trackIndex++) {
+    const textTrack = videoEl.textTracks[trackIndex];
+    if (textTrack.kind !== "metadata") {
+      continue;
+    }
+    textTrack.mode = "showing";
+    metaTextTrack = textTrack;
+    break;
+  }
+  if (!metaTextTrack) {
+    return;
+  }
+  const cuesLength = metaTextTrack.cues.length;
+  let cueIndex = 0;
+  while (cueIndex < cuesLength) {
+    const cue = metaTextTrack.cues[cueIndex];
+    if (cue.fired) {
+      cueIndex++;
+      continue;
+    }
+    const data = metadataPayloadParser(cue.value.data);
+    const programData = videoEl.getStartDate();
+    const startDate = data.start_date;
+    const endDate = data.end_date;
+    const startTime =
+      new Date(startDate) -
+      new Date(programData) -
+      videoEl.currentTime * 1000;
+    const duration = new Date(endDate) - new Date(startDate);
+    setTimeout(() => {
+      const toast = {
+        title: `Payload from timed Metadata ${data.payload}`,
+        duration: duration,
+      };
+      console.debug("Added toast ", JSON.stringify(toast));
+      ToastManager.addToast(toast);
+    }, startTime);
+    cue.fired = true;
+    cueIndex++;
+  }
+};
+if (Hls.isSupported()) {
+  /**
+   * initialize HLSController and add event listeners.
+   */
+  hlsController = new HLSController(hlsUrl, videoRef);
+  hlsStats = new HlsStats(hlsController.getHlsJsInstance(), videoEl);
+
+  hlsController.on(HLS_STREAM_NO_LONGER_LIVE, handleNoLongerLive);
+  hlsController.on(HLS_TIMED_METADATA_LOADED, metadataLoadedHandler);
+
+  hlsController.on(Hls.Events.MANIFEST_LOADED, manifestLoadedHandler);
+  hlsController.on(Hls.Events.LEVEL_UPDATED, levelUpdatedHandler);
+  hlsController.on(Hls.Events.ERROR, handleHLSError);
+} else if (videoEl.canPlayType("application/vnd.apple.mpegurl")) {
+    // Work where hls.js is not supported
+    videoEl.src = hlsUrl;
+    videoEl.addEventListener("timeupdate", handleTimeUpdateListener);
+    setIsMSENotSupported(true);
+}
+
+```
+
+>Please Note : Above implementation might change in future.
+
+## Limitations
+
+- It’s hard to sync metadata with stream exactly due to the latency from ffmpeg creating the stream. To address this, we’ve added 2s the actual `START-DATE` attribute.
```

Comment:
This is unclear - "To address this, we’ve added 2s the actual `START-DATE` attribute."